pulumitest/cleanup.go

```diff
@@ -15,3 +18,16 @@ var runningInCI = (func() func() bool {
 	_, ok := os.LookupEnv("CI")
 	return func() bool { return ok }
 })()
+
+var shouldRetainFilesOnFailure = (func() func() bool {
+	if value, ok := os.LookupEnv("PULUMITEST_RETAIN_FILES_ON_FAILURE"); ok {
+		if strings.EqualFold(value, "false") {
+			return func() bool { return false }
+		}
+		return func() bool { return true }
+	}
+	if skipDestroyOnFailure() {
+		return func() bool { return true }
+	}
+	return func() bool { return !runningInCI() }
+})()
```

Comment:
My understanding is that `shouldRetainFilesOnFailure` is a `func() bool`, where the `func` has a constant value. Why not just set `shouldRetainFilesOnFailure` to a var. If we want to make it immutable, we should put the "immutable" env vars into it's own package and then expose a getter function:

```go
// pulumitest/internal/env
package env
var shouldRetainFilesOnFailure bool

func init() {
    setShouldRetainFilesOnFailure()
}

func ShouldRetainFilesOnFailure() bool {
    return shouldRetainFilesOnFailure
}
```

The immediately invoked lambdas allow us to convert a block into an expression, but they introduce a lot of complexity and don't effect mutability.